controllers/snapshot/snapshot_adapter.go

```diff
@@ -566,8 +564,18 @@ func (a *Adapter) createCopyOfExistingEnvironment(existingEnvironment *applicati
 
 	err = a.client.Create(a.context, environment)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create environment %s: %w", environment.Name, err)
+		// We don't want to leave the new DTC on the cluster without the matching environment
+		dtcErr := a.client.Delete(a.context, dtc)
+		if dtcErr != nil {
+			a.logger.Error(err, "failed to create ephemeral environment", "environment.name", environment.Name)
+			return nil, fmt.Errorf("failed to delete dtc after creation of environment failed %s: %w", dtc.Name, dtcErr)
```

Comment:
Can we just log it as error and return only env creation error?


Or provide both errors
`failed to delete dtc %s: %w; failed to create env %s: %w", dtc.Name, dtcErr, env.Name, err`

Reply:
it should be also `DTC `

Reply:
Yes, good point, updated.